charts/nucliadb_node/templates/node.replicas.deploy.yaml

```diff
@@ -0,0 +1,170 @@
+{{- $release := .Release }}
+{{- $values := .Values }}
+{{- $chart := .Chart }}
+{{- range $i, $e := until (int .Values.replicaCount) }}
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: node-replicas-{{ $i }}
+  labels:
+    app: node-replicas-{{ $i }}
+    appType: node
+    app.kubernetes.io/name: node-replicas-{{ $i }}
+    app.kubernetes.io/instance: "{{ $release.Name }}"
+    app.kubernetes.io/managed-by: {{ $release.Service }}
+    version: "{{ $chart.Version | replace "+" "_" }}"
+    chart: "{{ $chart.Name }}"
+spec:
+  replicas: {{ $values.readReplicas.count }}
+  revisionHistoryLimit: {{ $values.revisionHistoryLimit }}
+  serviceName: node
+  selector:
+    matchLabels:
+      app: node-replicas-{{ $i }}
+      app.kubernetes.io/name: node-replicas-{{ $i }}
+      app.kubernetes.io/instance: "{{ $release.Name }}"
+      app.kubernetes.io/managed-by: {{ $release.Service }}
+  template:
+    metadata:
+      annotations:
+        # do not have access to dependency chart cm this component references
+        {{- if hasKey $values "extra_pod_annotations" }}
+{{ toYaml $values.extra_pod_annotations | indent 8 }}
+        {{- end }}
+      name: node-replicas-{{ $i }}
+      labels:
+        app: node-replicas-{{ $i }}
+        appType: node
+        app.kubernetes.io/name: node-replicas-{{ $i }}
+        app.kubernetes.io/instance: "{{ $release.Name }}"
+        app.kubernetes.io/managed-by: {{ $release.Service }}
+        version: "{{ $chart.Version | replace "+" "_" }}"
+        chart: "{{ $chart.Name }}"
+    spec:
+      nodeSelector:
+{{ toYaml $values.readReplicas.nodeSelector | indent 8 }}
+      topologySpreadConstraints:
+{{ toYaml $values.readReplicas.topologySpreadConstraints | indent 8 }}
+      affinity:
+{{ toYaml $values.readReplicas.affinity | indent 8 }}
+      tolerations:
+{{ toYaml $values.readReplicas.tolerations | indent 8 }}
+      dnsPolicy: ClusterFirst
+      volumes:
+      - name: data-dir
+        emptyDir:
+          sizeLimit: {{ mul $values.readReplicas.disk.numLocalDisks $values.readReplicas.disk.localDiskSize $values.readReplicas.disk.reservationPercentage | int }}Gi
+{{- if $values.nats.secretName }}
+      - name: nats-creds
+        secret:
+          secretName: {{ $values.nats.secretName }}
+{{- end }}
+      containers:
+      - name: writer
```

Comment:
I guess writer is still needed because we have all the health checking in the grpc server?

Reply:
Yes and the writer is what is connecting to the primary to sync the data as well